### DISCUSSION TITLE
Owncast 0.1.1: Unable to generate thumbnail: exit status 139 -fix?

### DISCUSSION THREAD

**Discussion by AtlasCentaurus** [owncast/owncast#3288] [Q&A] [answered]
Hi to everyone 

Today after doing a stream, i got the following errors:

**Segmentation faul** (only 1 time)
**Unable to generate thumbnail: exit status 139** (everytime)

Like an extra data: Today my Ubuntu version was upgraded from 22.04.3 LTS (Jammy Jellyfish) to  23.04 (Lunar Lobster)  and then I installed the ngix to get the ssl thing. The streams works very good, however it can not generate the thumbnail, does anyone knows how to fix this?

Thanks in advance

**Comment by gabek** [comment]
Do you have video passthrough enabled in your video encoding settings? What version of ffmpeg are you running? Could you see if there's any additional details you could get by running with the `--enableDebugFeatures` and `--enableVerboseLogging` flags?

**Reply by AtlasCentaurus** [reply]
Hi, Thank you for answering.

**Do you have video passthrough enabled in your video encoding settings?** 

No, I don't have video passthrough enabled in the encoding settings.

**What version of ffmpeg are you running?**

According to the info, it is the version 5.1.2-3ubuntu1 (take a look at the following info)

> ffmpeg -version
> ffmpeg version 5.1.2-3ubuntu1 Copyright (c) 2000-2022 the FFmpeg developers
> built with gcc 12 (Ubuntu 12.2.0-14ubuntu2)
> configuration: --prefix=/usr --extra-version=3ubuntu1 --toolchain=hardened --libdir=/usr/lib/x86_64-linux-gnu --incdir=/usr/include/x86_64-linux-gnu --arch=amd64 --enable-gpl --disable-stripping --enable-gnutls --enable-ladspa --enable-libaom --enable-libass --enable-libbluray --enable-libbs2b --enable-libcaca --enable-libcdio --enable-libcodec2 --enable-libdav1d --enable-libflite --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libglslang --enable-libgme --enable-libgsm --enable-libjack --enable-libmp3lame --enable-libmysofa --enable-libopenjpeg --enable-libopenmpt --enable-libopus --enable-libpulse --enable-librabbitmq --enable-librist --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libsrt --enable-libssh --enable-libsvtav1 --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx265 --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-lv2 --enable-omx --enable-openal --enable-opencl --enable-opengl --enable-sdl2 --disable-sndio --enable-libjxl --enable-pocketsphinx --enable-librsvg --enable-libmfx --enable-libdc1394 --enable-libdrm --enable-libiec61883 --enable-chromaprint --enable-frei0r --enable-libx264 --enable-libplacebo --enable-librav1e --enable-shared
> libavutil      57. 28.100 / 57. 28.100
> libavcodec     59. 37.100 / 59. 37.100
> libavformat    59. 27.100 / 59. 27.100
> libavdevice    59.  7.100 / 59.  7.100
> libavfilter     8. 44.100 /  8. 44.100
> libswscale      6.  7.100 /  6.  7.100
> libswresample   4.  7.100 /  4.  7.100
> libpostproc    56.  6.100 / 56.  6.100

>  fmpeg -hide_banner -codecs | grep 264
>  DEV.LS h264                 H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10 (decoders: h264 h264_v4l2m2m h264_qsv h264_cuvid ) (encoders: libx264 libx264rgb h264_nvenc h264_omx h264_qsv h264_v4l2m2m h264_vaapi )

 **--enableDebugFeatures**

> INFO[2023-08-31T11:20:20-06:00]/build/main.go:72 main.main() Owncast v0.1.1-linux-64bit (103d1684ca329cbffdaa66bc5a17b064a20c5942) 
> INFO[2023-08-31T11:20:20-06:00]/build/core/transcoder/transcoder.go:124 github.com/owncast/owncast/core/transcoder.(*Transcoder).Start() FFREPORT=file="data/logs/transcoder.log":level=32 ./ffmpeg -hide_banner -loglevel warning  -fflags +genpts -flags +cgop -i  ./data/tmp/offline.ts1440711066  -map v:0 -c:v:0 libx264 -b:v:0 5808k -maxrate:v:0 6272k -g:v:0 300 -keyint_min:v:0 300 -r:v:0 60 -x264-params:v:0 "scenecut=0:open_gop=0" -bufsize:v:0 6272k -profile:v:0 high -map a:0? -c:a:0 copy -sws_flags bilinear -filter:v:0 "scale=1600:900" -preset veryfast  -var_stream_map "v:0,a:0 " -f hls -hls_time 5 -hls_list_size 5 -hls_flags program_date_time+independent_segments -hls_playlist_type event -segment_format_options mpegts_flags=mpegts_copyts=1 -tune zerolatency -pix_fmt yuv420p -sc_threshold 0 -master_pl_name stream.m3u8 -hls_segment_filename http://127.0.0.1:45425/%v/stream-offline-%d.ts -max_muxing_queue_size 400 -method PUT http://127.0.0.1:45425/%v/stream.m3u8 
> INFO[2023-08-31T11:20:20-06:00]/build/router/router.go:439 github.com/owncast/owncast/router.Start() Web server is listening on port 8080.        
> INFO[2023-08-31T11:20:20-06:00]/build/router/router.go:441 github.com/owncast/owncast/router.Start() Configure this server by visiting /admin.    
> ERRO[2023-08-31T11:20:21-06:00]/build/core/transcoder/utils.go:92 github.com/owncast/owncast/core/transcoder.handleTranscoderMessage() Segmentation fault (core dumped)             
> ERRO[2023-08-31T11:20:21-06:00]/build/core/transcoder/transcoder.go:157 github.com/owncast/owncast/core/transcoder.(*Transcoder).Start() transcoding error. look at data/logs/transcoder.log to help debug. your copy of ffmpeg may not support your selected codec of libx264 https://owncast.online/docs/codecs/ 

**--enableVerboseLogging**

INFO[2023-08-31T11:25:29-06:00] Owncast v0.1.1-linux-64bit (103d1684ca329cbffdaa66bc5a17b064a20c5942) 
TRAC[2023-08-31T11:25:29-06:00] Creating webhooks table...                   
TRAC[2023-08-31T11:25:29-06:00] Creating users table...                      
TRAC[2023-08-31T11:25:29-06:00] http_listen_address sql: no rows in result set 
TRAC[2023-08-31T11:25:29-06:00] Resetting file directories to a clean slate. 
TRAC[2023-08-31T11:25:29-06:00] Cleaning data/hls                            
TRAC[2023-08-31T11:25:29-06:00] Warming config value cache                   
TRAC[2023-08-31T11:25:29-06:00] Transcoder response service listening on: 33257 
TRAC[2023-08-31T11:25:29-06:00] Firing transcoder with offline stream state  
TRAC[2023-08-31T11:25:29-06:00] Cleaning data/hls                            
TRAC[2023-08-31T11:25:29-06:00] Max process connection count changed from system limit of 1024 to 734003 
TRAC[2023-08-31T11:25:29-06:00] Chat server started with max connection count of 734003 
TRAC[2023-08-31T11:25:29-06:00] Removing chat messages older than 2 hours    
TRAC[2023-08-31T11:25:29-06:00] Creating federation followers table...       
DEBU[2023-08-31T11:25:29-06:00] Started Webhook worker 6                     
DEBU[2023-08-31T11:25:29-06:00] Started Webhook worker 1                     
TRAC[2023-08-31T11:25:29-06:00] RTMP server is listening for incoming stream on port: 1935 
DEBU[2023-08-31T11:25:29-06:00] Started Webhook worker 2                     
DEBU[2023-08-31T11:25:29-06:00] Started Webhook worker 3                     
DEBU[2023-08-31T11:25:29-06:00] Started Webhook worker 4                     
DEBU[2023-08-31T11:25:29-06:00] Started Webhook worker 5                     
TRAC[2023-08-31T11:25:29-06:00] Creating federation followers table...       
TRAC[2023-08-31T11:25:29-06:00] Creating federation outbox table...          
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 1                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 1                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 6                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 2                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 5                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 4                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 2                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 4                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 3                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 5                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 3                 
DEBU[2023-08-31T11:25:29-06:00] Started ActivityPub worker 6                 
INFO[2023-08-31T11:25:29-06:00] Web server is listening on port 8080.        
INFO[2023-08-31T11:25:29-06:00] Configure this server by visiting /admin.    
DEBU[2023-08-31T11:25:29-06:00] Segmentation fault (core dumped)             
ERRO[2023-08-31T11:25:29-06:00] Segmentation fault (core dumped)             
ERRO[2023-08-31T11:25:29-06:00] transcoding error. look at data/logs/transcoder.log to help debug. your copy of ffmpeg may not support your selected codec of libx264 https://owncast.online/docs/codecs/

**Comment by gabek** [comment]
> ERRO[2023-08-31T11:25:29-06:00] transcoding error. look at data/logs/transcoder.log to help debug. 

Could you take a look at that file and see if there's anything helpful?

**Reply by AtlasCentaurus** [reply]
this is what the transcoder log from the data folder shows:

> ffmpeg started on 2023-08-31 at 11:45:18
> Report written to "data/logs/transcoder.log"
> Log level: 32
> Command line:
> ./ffmpeg -hide_banner -loglevel warning -fflags +genpts -flags +cgop -i ./data/tmp/offline.ts1200562468 -map v:0 -c:v:0 libx264 -b:v:0 5808k -maxrate:v:0 6272k -g:v:0 300 -keyint_min:v:0 300 -r:v:0 60 -x264-params:v:0 "scenecut=0:open_gop=0" -bufsize:v:0 6272k -profile:v:0 high -map "a:0?" -c:a:0 copy -sws_flags bilinear -filter:v:0 "scale=1600:900" -preset veryfast -var_stream_map "v:0,a:0 " -f hls -hls_time 5 -hls_list_size 5 -hls_flags program_date_time+independent_segments -hls_playlist_type event -segment_format_options "mpegts_flags=mpegts_copyts=1" -tune zerolatency -pix_fmt yuv420p -sc_threshold 0 -master_pl_name stream.m3u8 -hls_segment_filename "http://127.0.0.1:35319/%v/stream-offline-%d.ts" -max_muxing_queue_size 400 -method PUT "http://127.0.0.1:35319/%v/stream.m3u8"

**Reply by AtlasCentaurus** [reply, marked as answer]
Hello!

After one day of searching for solutions without find something accurate, my conclusion is that this trouble came from others variants and not from the owncast software itself and that such a trouble could have different reasons to happen.

My Guess

I think that what caused this trouble was the upgrade of my Ubuntu installation going from the version 22.04.3 TLS to the version 23.04,  in the tehory, everything should work fine after the upgrade but unfortunately the owncast was not working well after upgrading.

To me... this issue happened due to 1 of the following 2 things:

1.- Most of the Ubuntu packets were broken or corrupted after the upgrade. (doing the recovery mode to fix them did not help)

2.-  Uncompatibility with the Ubuntu version 23.04. (I am not sure of this, y think someone should make a test with a clean installation of this version, due to the lack of time, i couldn't make it... sorry.)

What was the solution?

Essentially... I took the most basic and common action to resolve this by doing a clean installation of the Ubuntu going back to use the version 22.04.3 TLS, then doing a clean installation of the Owncast re-configuring everything.

Now the Owncast is working smooth and like should be.

Thank You so much for your kind attention and for the answers.
